src/AppBundle/Utils/ListUtils/AbstractList.php

```diff
@@ -0,0 +1,310 @@
+    }
+
+    /**
+     * @return \Twig_Environment
+     */
+    protected function getTwig()
+    {
+        return $this->twig;
+    }
+
+    /**
+     * @return Router
+     */
+    protected function getRouter()
+    {
+        return $this->router;
+    }
+
+    /**
+     * @param $attributes
+     * @param null $object
+     * @return bool
+     */
+    public function isGranted($attributes, $object = null)
+    {
+        return $this->checker->isGranted($attributes, $object);
+    }
+
+    /**
+     * @return Array
+     */
+    public function getItems()
+    {
+        return $this->items;
+    }
+
+    /**
+     * @param Array $items
+     */
+    protected function setItems($items)
+    {
+        $this->items = $items;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     */
+    protected function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isDatatable()
+    {
+        return $this->datatable;
+    }
+
+    /**
+     * @param boolean $datatable
+     */
+    protected function setDatatable($datatable)
+    {
+        $this->datatable = $datatable;
+    }
+
+    public function getColumns()
+    {
+        return $this->columns;
+    }
+
+
+    protected function addColumn(Column $col)
+    {
+        $this->columns->add($col);
+        return $this;
+    }
+
+    protected function addActionLine(ActionLine $action)
+    {
+        $this->actionsLine->add($action);
+    }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getActionsLine()
+    {
+        return $this->actionsLine;
+    }
+
+    protected function addActionList(ActionList $action)
+    {
+        $this->actionsList->add($action);
+    }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getActionsList()
+    {
+        return $this->actionsList;
+    }
+
+    public function getRowId($item)
+    {
+        /* Oui, oui, ca parait bizarre mais ca marche!! */
+        $function = $this->itemIdAccessor;
+        return $function($item);
+    }
+
+    public function getUrl()
+    {
+        return $this->url;
+    }
+
+    /**
+     * @param string $url
+     * @return string
+     */
+    protected function setUrl($url)
+    {
+        $this->url = $url;
+    }
+
+    /**
+     * @return array
+     */
+    public function getExportFormats()
+    {
+        return $this->export_formats;
+    }
+
+    public function hasExportFormats()
+    {
+        return !$this->export_formats->isEmpty();
+    }
+
+    protected function addExportFormats($format,$label)
+    {
+        if(
+            ($format == self::FORMAT_EXPORT_CSV)    ||
+            ($format == self::FORMAT_EXPORT_XLSX)
+        )
+        {
+            $this->export_formats->set($format,$label);
+        }
+    }
+
+    /**
+     * @return String
+     */
+    public function getCssClass()
```

Comment:
Ca devrait plutot être un array, du genre addCssClass($class) et à la fin on fait un implode(' ', $this->cssClass), ca permet d'en ajouter à plusieurs endroits